src/main/java/web/HomePage.java

```diff
@@ -0,0 +1,83 @@
+package web;
+
+import com.zebrunner.carina.webdriver.decorator.ExtendedWebElement;
+import com.zebrunner.carina.webdriver.decorator.PageOpeningStrategy;
+import com.zebrunner.carina.webdriver.gui.AbstractPage;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.support.FindBy;
+import web.components.LoginModal;
+import web.components.NavBar;
+import web.components.ProductCard;
+import web.components.ProductCategory;
+
+public class HomePage extends AbstractPage {
+
+    @FindBy(xpath = "//button[@id='next2']")
+    private ExtendedWebElement nextPageButton;
+
+    @FindBy(xpath = "//button[@id='prev2']")
+    private ExtendedWebElement prevPageButton;
+
+    @FindBy(xpath = "//*[@class='card-block']")
+    private ExtendedWebElement cardLoadedMarker;
+
+    @FindBy(xpath = "//div[contains(@class, 'card ')]")
+    private List<ProductCard> productList;
+
+    @FindBy(xpath = "//a[@id = 'itemc']")
+    private List<ProductCategory> categories;
+
+    @FindBy(xpath = "//div[contains(@class, 'card ')]")
+    private List<ProductCard> productCards;
+
+    public HomePage(WebDriver driver) {
+        super(driver);
+        setPageOpeningStrategy(PageOpeningStrategy.BY_ELEMENT);
+        setUiLoadedMarker(cardLoadedMarker);
+    }
+
+    public void clickNextButton() {
+        nextPageButton.click();
+    }
+
+    public void clickPrevButton() {
+        prevPageButton.click();
+    }
+
+    public List<ProductCard> getDisplayedProducts() {
+        waitForJSToLoad(250L);
```

Comment:
When navigating the products list by pressing the next and previous buttons, a jquery script is called to fetch the products and modify the DOM element.  The page dynamically loads the products instead of going to a new page/new url. In this case,  isPageOpen does not work and the element is always present.